src/config.rs

```diff
@@ -96,17 +96,20 @@ impl<T: Into<String>> From<T> for OverrideFile {
     }
 }
 
+// Represents the reason why the active toolchain is active.
 #[derive(Debug)]
-pub(crate) enum OverrideReason {
+pub(crate) enum ActiveReason {
```

Comment:
This rename should be in a separate commit.

Reply:
@djc I'm not so sure about that... Although it could possibly make the PR easier to review, the intermediate state that you're looking for might have inconsistent semantics.

Reply:
I generally don't think this is a big issue? Like, it's only within the context of a few commits (that are part of the same PR), after which the branch is in a consistent state again.

Reply:
My instinct is that one semantic change should be in one commit, the code should compile and pass tests at each commit, and that artificially small commits aren't helpful. That's what I'm used to from my experience working on different projects. I don't really see any value in having the branch in an inconsistent state to begin with.